src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs

```diff
@@ -713,7 +713,7 @@ public int GetDocumentCount(Solution solution)
                 return count;
             }
 
-            public IEnumerable<Document> GetDocuments(Solution solution)
+            public IEnumerable<Document> TryGetDocuments(Solution solution)
```

Comment:
this is a bit wacky as 'try' generally makes us think you can get Null back.  Not that this might return 'empty'.

Note: is it jsut "broken" if you ask for the docs for a solution that isn't the solution-id we're the work coordinator for?

Reply:
implementation wise, re-analyze request made for solution "A" and put in the queue, and when it actually have time to do work and solution "A" is already gone and solution "B" is now current, so it returning 0 seems completely fine.

Reply:
Agreed: I'd expect null here. `TryGetDocumentsOrEmpty` so it's actually clear? Or minimally stick an XML doc comment on it.

Reply:
@CyrusNajmabadi @jasonmalinowski That is more feedback for #27833. I'll file a new issue for the remaining items to be addressed after this is merged.

Reply:
I guess my concern is this is more confusing IMHO than it started with. :smile: